src/jwt.cc

```diff
@@ -23,6 +24,18 @@
 namespace google {
 namespace jwt_verify {
 
+namespace {
+
+static absl::flat_hash_set<std::string> implemented_algs = {
```

Comment:
per google style guide.  const variable is camel case prefixed with k.  as:
kImplementedAlgs

Reply:
please also add "const" for it as
static const absl:...

Reply:
Fixed in e9757de!

Reply:
How about the constants in the test cases, like `Ps{256,384,512}JwtTextWithCorrectKid` or `PublicKeyRSAPSS`? I took reference from the existing test cases. I could change them all to line up with the style guide too.

Reply:
I mixed two coding styles: Envoy and Google.  Envoy is just camel case without prefix "k", and Google has prefix "k".  
Then, please remove prefix "k", and we use Envoy style here in this library.

Reply:
I just noticed this in the [envoy style guide](https://github.com/envoyproxy/envoy/blob/master/STYLE.md#:~:non-PoD%20static%20and%20global%20variables%20are%20forbidden) about static global variables. I don't think we're affected by the static initialization order problem, since `ImplementedAlgs` is in an anonymous namespace and won't be used by other files.

What do you think of `ImplementedArgs`, in relation to the static initialization order problem?

Reply:
Found this. https://google.github.io/styleguide/cppguide.html#:~:text=Maps,%20sets,%20and%20other%20dynamic%20containers

Will take a look at `absl/algorithm/container.h`

Reply:
We do have a initialization order issue.  Please take a look Envoy Singleton implementation,  may have to add a static function, and new such map inside the static function in the first call to make sure it is not initialized  before main()

Reply:
What do you think of moving the global `static const flat_hash_set` into a function instead? (9cb97fe) This has the additional benefit of encapsulating the checking in a nice function.

Also, I'm new to C++. Help me understand the initialization order problem! My understanding is that there is a problem only if there is more than one compilation unit that uses the same global variable that has a static storage duration. Specifically for this case, since `ImplementedAlgs` is in an anonymous namespace usable only in this file, it is only available in this compilation unit, and so I thought that specifically for this case there wouldn't be a problem.

I do understand that we might not want to rely on this though, in case someone in future accidentally decides to use `ImplementedAlgs` out of the anonymous namespace.

I also explored one of the common patterns suggested in the Google style guide:

```
TEST(Foo, Bar) {
  std::string needle = "foo";
  std::array<const char*, 3> haystack = {"foo", "bar", "baz"};
  EXPECT_TRUE(std::is_trivially_destructible<decltype(haystack)>::value);
  EXPECT_TRUE(std::find(haystack.begin(), haystack.end(), needle) != haystack.end());
}
```

We could use a static const trivially destructible array instead, which would avoid the initialization order problem, but it seems like in order to use `std::find` (and `haystack.begin()` and `haystack.end()`) I'd have to specify the size of the array manually.

Is the Envoy Singleton implementation this one? https://github.com/envoyproxy/envoy/blob/master/include/envoy/singleton/manager.h

Reply:
It is here

https://github.com/envoyproxy/envoy/blob/master/source/common/singleton/const_singleton.h

Reply:
Initialization order is very complicated issue,  I don't fully understand it.  Basically, you want to initialize your data after main(), not before.   If you create a global T, it is before main(),   if you new it in a function as static it is after.  That is all I need to know

Reply:
Thanks! Okay after referring to the envoy one, I think this implementation is serves the same purpose, unless you're intending for `ImplementedAlgs` to be accessed as-is, as a `flat_hash_set`?